src/fs.rs

```diff
@@ -358,3 +463,174 @@ impl<D: Drive + Clone> Future for Create<D> {
         Poll::Ready(Ok(File::from_fd(fd, driver)))
     }
 }
+
+/// A future representing an syncing a file.
+pub struct SyncAll<'a, D: Drive> {
+    file: Pin<&'a mut File<D>>,
+}
+
+impl<'a, D: Drive + Clone> Future for SyncAll<'a, D> {
+    type Output = io::Result<()>;
+
+    fn poll(mut self: Pin<&mut Self>, ctx: &mut Context<'_>) -> Poll<io::Result<()>> {
+        self.file.as_mut().poll_sync_all(ctx)
+    }
+}
+
+/// A future representing an syncing a file.
+pub struct SyncData<'a, D: Drive> {
+    file: Pin<&'a mut File<D>>,
+}
+
+impl<'a, D: Drive + Clone> Future for SyncData<'a, D> {
+    type Output = io::Result<()>;
+
+    fn poll(mut self: Pin<&mut Self>, ctx: &mut Context<'_>) -> Poll<io::Result<()>> {
+        self.file.as_mut().poll_sync_data(ctx)
+    }
+}
+
+/// A future representing an syncing a file.
+pub struct MetadataFut<'a, D: Drive> {
+    file: Pin<&'a mut File<D>>,
+}
+
+impl<'a, D: Drive + Clone> Future for MetadataFut<'a, D> {
+    type Output = io::Result<Metadata>;
+
+    fn poll(mut self: Pin<&mut Self>, ctx: &mut Context<'_>) -> Poll<io::Result<Metadata>> {
+        self.file.as_mut().poll_metadata(ctx)
+    }
+}
+
+/// Metadata information about a file.
+#[derive(Clone)]
+pub struct Metadata {
+    stat: libc::statx,
+}
+
+impl Metadata {
+    /// Returns the file type for this metadata.
+    pub fn file_type(&self) -> FileType {
+        FileType(self.stat.stx_mode as libc::mode_t)
+    }
+
+    /// Returns `true` if this metadata is for a directory. The
+    /// result is mutually exclusive to the result of
+    /// [`Metadata::is_file`], and will be false for symlink metadata
+    /// obtained from [`symlink_metadata`].
+    pub fn is_dir(&self) -> bool {
+        self.file_type().is_dir()
+    }
+
+    /// Returns `true` if this metadata is for a regular file. The
+    /// result is mutually exclusive to the result of
+    /// [`Metadata::is_dir`], and will be false for symlink metadata
+    /// obtained from [`symlink_metadata`].
+    pub fn is_file(&self) -> bool {
+        self.file_type().is_file()
+    }
+
+    /// Returns the size of the file, in bytes, this metadata is for.
+    pub fn len(&self) -> u64 {
+        self.stat.stx_size
+    }
+
+    /// Returns the permissions of the file this metadata is for.
+    pub fn permissions(&self) -> Permissions {
+        Permissions(self.stat.stx_mode as libc::mode_t)
+    }
+
+    /// Returns the last modification time listed in this metadata.
+    pub fn modified(&self) -> io::Result<libc::statx_timestamp> {
```

Comment:
Rust doesn't allow to construct `time::SystemTime` manually, so I stuck the original type for now. Any better ideas would be appreciated.

Reply:
One way to do this is to convert the timestamp with [`Duration::new`](https://doc.rust-lang.org/stable/std/time/struct.Duration.html#method.new) and add it to [`SystemTime::UNIX_EPOCH`](https://doc.rust-lang.org/stable/std/time/struct.SystemTime.html#associatedconstant.UNIX_EPOCH) with  [`checked_add`](https://doc.rust-lang.org/stable/std/time/struct.SystemTime.html#method.checked_add).

Reply:
good idea, thank you, I will try taht